lib/optimize/ConcatenatedModule.js

```diff
@@ -829,8 +829,6 @@ class ConcatenatedModule extends Module {
 						topLevelDeclarations.add(decl);
 					}
 				}
-			} else {
-				this.buildInfo.topLevelDeclarations = undefined;
```

Comment:
removing that is not correct.

When at least one module doesn't provide topLevelDeclarations we can't tell about the topLevelDeclarations of the concatenated module, so it should be `undefined` (which means "no information")